janusgraph-codepipelines-ci/README.md

```diff
@@ -0,0 +1,83 @@
+# JanusGraph CodePipelines CI
+CodePipelines CI is a mechanism JanusGraph can use to do release testing in massively
+parallel fashion (to the extent of AWS CodePipelines and CodeBuild service limits).
+
+## Prerequisites
+This procedure requires you to have an AWS account.
+It also requires you to create two service roles in IAM: one for CodePipeline and
+one for CodeBuild. Finally, you need to have the [AWS CLI](https://aws.amazon.com/cli/) installed and on your path.
+
+1. Get a personal access token from [GitHub](https://github.com/settings/tokens) with `repo` and `admin:repo_hook` scopes.
+The `repo` scope is used to push the latest updates on the branch selected below. The `admin:repo_hook` scope is for
+setting up the post-commit hook on GitHub programmatically.
+2. Navigate to the [AWS Console](https://console.aws.amazon.com) and
+[create an IAM User](http://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html) with the following managed
+policies: AmazonS3FullAccess, AWSCodePipelineFullAccess, AWSCodeBuildAdminAccess. This user should be created for
+__Programmatic access__.
+3. For this user, create security credentials and then register them in the `code-pipelines` profile on your computer
+with `aws configure --profile code-pipelines`. Create this profile with a default region that
+[supports CodePipelines and CodeBuild](https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services/)
+and a default [output format](http://docs.aws.amazon.com/cli/latest/userguide/controlling-output.html#controlling-output-format)
+of your choice (`json`, `text`, and `table` are available).
+4. Create a service role for CodeBuild as described
+[here](http://docs.aws.amazon.com/codebuild/latest/userguide/setting-up.html#setting-up-service-role).
+5. Create an IAM policy for a CodePipelines service role as described
+[here](http://docs.aws.amazon.com/codepipeline/latest/userguide/iam-identity-based-access-control.html#view-default-service-role-policy).
+6. Create a service role for CodePipelines by using the policy you created in step 5 and then set the STS trust to
+codepipeline.amazonaws.com, like what you did for CodeBuild.
+
+## Testing
+Follow the three steps below to build and run the program to create the stack and start a build that runs regular and
+TinkerPop tests in parallel.
+1. First, clean up and set some constant environment variables.
+
+```bash
+mvn clean package
+#constants
+export AWS_PROFILE_NAME='code-pipelines'
+export AWS_ACCOUNT_NUMBER=`aws sts get-caller-identity --output text | cut -f1`
+```
+
+2. Second, configure the parameters below to customize your test stack.
+
+```bash
+#define variables
+#GitHub personal access token you created in step 1
+export GITHUB_TOKEN=''
+#GitHub organization or user name
```

Comment:
Add "of the repository to build". Need to make it clear that it's the user/group of the repository to test not (necessarily) the GitHub handle.